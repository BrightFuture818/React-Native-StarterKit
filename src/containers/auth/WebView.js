/**
 * Auth WebView
 *  - Used by sign up and password reset
 *  - Shows the WebViews and pops when finished
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component, PropTypes } from 'react';
import { Actions } from 'react-native-router-flux';

// Components
import WebView from '@components/general/WebView';

let timeout;

/* Component ==================================================================== */
class AuthWebView extends Component {
  static componentName = 'AuthWebView';

  static propTypes = {
    url: PropTypes.string.isRequired,
  }

  componentWillUnmount = () => {
    // Clear the timeout, otherwise we'll get warnings
    // when the user unmounts mid-way timeout
    clearTimeout(timeout);
  }

  /**
    * Pop back a scene when URL changes
    * + for when the action is completed
    * + restricting people from freely browsing
    */
  urlChanged = (newUrl) => {
    if (newUrl !== this.props.url) {
      timeout = setTimeout(() => {
        clearTimeout(timeout);
        Actions.pop();
      }, 2000);
    }
  }

  render = () => (
    <WebView
      onNavigationStateChange={this.urlChanged}
      url={this.props.url}
    />
  )
}

/* Export Component ==================================================================== */
export default AuthWebView;
