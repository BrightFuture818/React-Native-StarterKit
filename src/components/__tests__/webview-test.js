/**
 * Test to check if the component renderes correctly
 */
 /* global it expect */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import WebView from '../general/WebView';

/**
 * Check if alerts renders correcly
 * and asserting it to the matching snapshot
 */
it('WebView renders correcly', () => {
  const tree = renderer.create(
    <WebView url={'http://google.com'} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
