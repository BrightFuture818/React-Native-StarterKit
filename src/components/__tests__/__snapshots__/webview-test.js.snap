exports[`test WebView renders correcly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }>
  <ActivityIndicator
    animating={true}
    color="#AAA"
    hidesWhenStopped={true}
    size="large" />
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    } />
</View>
`;
